### ISSUE TITLE
100% cpu on endless backfill loop when moving from grid back to single room

### ISSUE THREAD

**Issue by bwindels** [element-hq/hydrogen-web#160]
Happened on (Webkit) Edge in PWA mode, closing the grid with a room selected.

loop is interrupted and cpu goes back to normal by switching room
[loadtop-cpu-spin-edge-pwa.zip](https://github.com/vector-im/hydrogen-web/files/5391453/loadtop-cpu-spin-edge-pwa.zip)

**Comment by bwindels** [comment]
I think this happens in rooms that don't have enough events to fill the viewport.

**Comment by bwindels** [comment]
So, I have been able to reproduce this reliably. At least I think it is the same issue.

So this is a login on an account with only 2 rooms, each with less than 10 events in their timeline. Both rooms have a prev_batch in their room response even though there is nothing to paginate back to, but ok.
Now, when opening the timeline, it tries to paginate and does `/messages?dir=b` with this token. The HS returns me the same messages as in the initial sync (but reversed). So it basically did a forward fill, sort off.

Hydrogen detects that all events are overlapping and ends up linking the fragment with itself, setting `previousId` to `0`, the same as its `id`. Now when trying to load more messages, it enters in an infinite loop.

I've captured the server behaviour in a HAR file.

[pagination-edge.zip](https://github.com/vector-im/hydrogen-web/files/6070637/pagination-edge.zip)
